Added reference kernel fallback for conv

lamarrr opened the pull request:
I added the reference kernel fallback for the conv's EvalQuantizedPerChannel routine, as suggested by @freddan80

lamarrr committed:
Added reference kernel fallback for conv

I added the reference kernel fallback for the conv's EvalQuantizedPerChannel routine, as suggested by @freddan80

freddan80 reviewed (approved):
Thanks for your contribution @lamarrr!

freddan80 commented on tensorflow/lite/experimental/micro/kernels/cmsis-nn/conv.cc:
```diff
@@ -235,7 +236,15 @@ TfLiteStatus EvalQuantizedPerChannel(
   }
 #else
-#error ARM_MATH_DSP and ARM_MATH_LOOPUNROLL must be set
+
```
Actually, it would be good for the user to be notified that there's no optimization for this target. You can add a directive for that. For example:
#pragma message("CMSIS-NN optimization for conv not available for this target. Using ref kernel.")

lamarrr replied:
Great idea!

lamarrr replied:
I'm thinking it'd be better to have a more descriptive notification. i.e:

"CMSIS-NN optimization for _conv_ not available for this target (ARM_MATH_DSP and ARM_MATH_LOOPUNROLL not set). Using reference kernel"

what do you say?

lamarrr replied:
done!

lamarrr committed:
added compile time notification message

Added the pragma message directive to notify the user of the absence of CMSIS-NN library

rthadur commented:
Gentle ping @petewarden for review ?
